test/dx_mgn_pool.js

```diff
@@ -296,5 +311,43 @@ contract("DxMgnPool", (accounts) => {
 
       await truffleAssert.reverts(instance.withdrawMagnolia(), "MGN has not been unlocked, yet")
     })
+    it("withdrawals the right amount of MGN", async () => {
+      const depositTokenMock = await MockContract.new()
+      const secondaryTokenMock = await MockContract.new()
+      const mgnTokenMock = await MockContract.new()
+      const mgnToken = await TokenFRT.deployed()
+      const dxMock = await MockContract.new()
+      const dx = await DutchExchange.new()
+      const poolingEndBlock = (await web3.eth.getBlockNumber()) + 100
+      const instance = await DxMgnPool.new(depositTokenMock.address, secondaryTokenMock.address, mgnTokenMock.address, dxMock.address, poolingEndBlock)
+      
+
+      await depositTokenMock.givenAnyReturnBool(true)
+      await instance.deposit(10)
+
+      await dxMock.givenAnyReturnUint(10)
+      const postSellOrder = dx.contract.methods.postSellOrder(accounts[0], accounts[0], 0, 0).encodeABI()
+      const reponseType = (abi.rawEncode(["uint", "uint"], [2, 0]))
+      await dxMock.givenMethodReturn(postSellOrder, reponseType)
+      await instance.participateInAuction()
+
+      await waitForNBlocks(101, accounts[0])
+
+      const balanceOf = mgnToken.contract.methods.balanceOf(accounts[0]).encodeABI()
+      await mgnTokenMock.givenMethodReturnUint(balanceOf, 100)      
+
+      await instance.withdrawMGN()
+      
+      assert.equal(await instance.totalMgn.call(), 100)
+
+      await depositTokenMock.givenAnyReturnBool(true)
```

Comment:
You already have `depositTokenMock.givenAnyReturnBool(true)` called above. No need to call it again.